platform/ios/src/MGLMapView.mm

```diff
@@ -4547,7 +4547,13 @@ - (void)updateAnnotationViews
         
         MGLAnnotationContext &annotationContext = pair.second;
         MGLAnnotationView *annotationView = annotationContext.annotationView;
-       
+
+        // Defer to the shape/polygon styling delegate methods
+        if ([annotationContext.annotation isKindOfClass:[MGLMultiPoint class]])
```

Comment:
You could instead check for conformance to MGLShape if you're concerned about future class hierarchy chances.